modules/api/pkg/resources/machine/machinedeployment.go

```diff
@@ -444,3 +452,51 @@ func Validate(nd *apiv1.NodeDeployment, controlPlaneVersion *semverlib.Version)
 
 	return nd, nil
 }
+
+// validateAutoUpdateMDEnforcement validates if auto-update settings of node deployment are aligned with the
+// admin settings of machine deployment auto updates.
+func validateAutoUpdateMDEnforcement(ctx context.Context, nd *apiv1.NodeDeployment, settingsProvider provider.SettingsProvider) error {
+	settings, err := settingsProvider.GetGlobalSettings(ctx)
+
+	if err != nil {
+		return common.KubernetesErrorToHTTPError(err)
+	}
+
+	mdOptions := settings.Spec.MachineDeploymentOptions
+
+	if mdOptions.AutoUpdatesEnabled && mdOptions.AutoUpdatesEnforced {
+		errorMessage := errors.New("auto update cannot be disabled because it has been enforced by the admin")
+		osSpec := nd.Spec.Template.OperatingSystem
+		// OS specifics
+		switch {
+		case osSpec.Ubuntu != nil:
+			if !osSpec.Ubuntu.DistUpgradeOnBoot {
+				return errorMessage
+			}
+		case osSpec.CentOS != nil:
+			if !osSpec.CentOS.DistUpgradeOnBoot {
+				return errorMessage
+			}
+		case osSpec.RHEL != nil:
+			if !osSpec.RHEL.DistUpgradeOnBoot {
+				return errorMessage
+			}
+		case osSpec.Flatcar != nil:
+			if !osSpec.Flatcar.DisableAutoUpdate {
+				return errorMessage
+			}
```

Comment:
We probably need to reverse the if check here

Reply:
Yea, good catch 😉  I had it correct initially but then flipped it during refactoring 😵‍💫